File: MagicCollectionSpring/src/main/java/pga/magiccollectionspring/inventory/domain/enums/Language.java
package pga.magiccollectionspring.inventory.domain.enums;

public enum Language {
    ENGLISH("EN", "Ingles"),
    SPANISH("ES", "Espanol"),
    FRENCH("FR", "Frances"),
    GERMAN("DE", "Aleman"),
    ITALIAN("IT", "Italiano"),
    PORTUGUESE("PT", "Portugues"),
    JAPANESE("JP", "Japones"),
    CHINESE("CN", "Chino"),
    RUSSIAN("RU", "Ruso"),
    KOREAN("KR", "Coreano");

    private final String code;
    private final String spanishName;

    Language(String code, String spanishName) {
        this.code = code;
        this.spanishName = spanishName;
    }

    public String getCode() { return code; }
    public String getSpanishName() { return spanishName; }

    public static Language fromCode(String text) {
        if (text == null) throw new IllegalArgumentException("El idioma no puede estar vacio");
        for (Language l : Language.values()) {
            if (l.code.equalsIgnoreCase(text) || l.name().equalsIgnoreCase(text) || l.spanishName.equalsIgnoreCase(text)) {
                return l;
            }
        }
        throw new IllegalArgumentException("Idioma no soportado: " + text);
    }
}

File: MagicCollectionAndroid/app/src/main/java/com/pga/magiccollection/domain/model/enums/Language.kt
package com.pga.magiccollection.domain.model.enums

enum class Language(val code: String, val displayName: String) {
    ENGLISH("EN", "English"),
    SPANISH("ES", "Spanish"),
    FRENCH("FR", "French"),
    GERMAN("DE", "German"),
    ITALIAN("IT", "Italian"),
    PORTUGUESE("PT", "Portuguese"),
    JAPANESE("JP", "Japanese"),
    CHINESE("CN", "Chinese"),
    RUSSIAN("RU", "Russian"),
    KOREAN("KR", "Korean");

    companion object {
        fun fromCode(text: String?): Language {
            if (text == null) throw IllegalArgumentException("Language cannot be empty")
            return values().find {
                it.code.equals(text, ignoreCase = true) ||
                        it.name.equals(text, ignoreCase = true) ||
                        it.displayName.equals(text, ignoreCase = true)
            } ?: throw IllegalArgumentException("Unsupported language: $text")
        }
    }
}